File: src/main/resources/templates/php/solution/BubbleSort.php
<?php

class BubbleSort implements SortStrategy {

    /**
     * Sorts dates with BubbleSort.
     *
     * @param array $input the array of Dates to be sorted
     */
    public function performSort(array &$input) {
        $n = count($input);
        for ($i = $n - 1; $i >= 0; $i--) {
            for ($j = 0; $j < $i; $j++) {
                if ($input[$j] > $input[$j + 1]) {
                    $temp = $input[$j];
                    $input[$j] = $input[$j + 1];
                    $input[$j + 1] = $temp;
                }
            }
        }
    }
}

File: src/main/resources/templates/cpp/solution/BubbleSort.cpp
#include "BubbleSort.h"

/**
* Sorts integers with BubbleSort.
*
* @param std::vector<int> &input the vector of integers to be sorted
*/
void BubbleSort::perform_sort(std::vector<int> &input) {
    int n = input.size();
    for (int i = n - 1; i >= 0; i--) {
        for (int j = 0; j < i; j++) {
            if (input[j] > input[j + 1]) {
                std::swap(input[j], input[j + 1]);
            }
        }
    }
}